From b13b7e38b38e2c79c0a13be32e05f0451483cd01 Mon Sep 17 00:00:00 2001
From: James Hilliard <james.hilliard1@gmail.com>
Date: Sat, 12 Jun 2021 15:32:37 -0600
Subject: [PATCH] Fix duplicate xbr packages and xbr templates package_data.
 (#1492)

Signed-off-by: James Hilliard <james.hilliard1@gmail.com>
[james.hilliard1@gmail.com: backport from upstream commit
a370f9e7154e59b313cb675bfcaf8cad6011bd52]
---
 MANIFEST.in |  1 -
 setup.py    | 10 ++++++++--
 2 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/MANIFEST.in b/MANIFEST.in
index f4cb6da1..1dd32f0e 100644
--- a/MANIFEST.in
+++ b/MANIFEST.in
@@ -1,4 +1,3 @@
 include LICENSE
 include autobahn/nvx/_utf8validator.c
 recursive-include autobahn/wamp/gen/schema *
-recursive-include autobahn/xbr/templates *
diff --git a/setup.py b/setup.py
index cf4e29ba..f27006da 100644
--- a/setup.py
+++ b/setup.py
@@ -166,6 +166,12 @@ packages = [
     'twisted.plugins',
 ]
 
+xbr_packages = [
+    'autobahn.xbr',
+    'autobahn.asyncio.xbr',
+    'autobahn.twisted.xbr',
+]
+
 package_data = {'autobahn.asyncio': ['./test/*']}
 
 entry_points = {
@@ -179,8 +185,8 @@ if 'AUTOBAHN_STRIP_XBR' in os.environ:
     shutil.rmtree('autobahn.egg-info', ignore_errors=True)
 else:
     extras_require_all += extras_require_xbr
-    packages += ['autobahn.xbr', 'autobahn.asyncio.xbr', 'autobahn.twisted.xbr']
-    package_data['xbr'] = ['./xbr/contracts/*.json']
+    packages += xbr_packages
+    package_data['xbr'] = ['./xbr/templates/*.py.jinja2']
     entry_points['console_scripts'] += ["xbrnetwork = autobahn.xbr._cli:_main"]
 
 # development dependencies
-- 
2.25.1

